compiler/fir/analysis-tests/testData/resolve/cfg/selfTypes.kt

```diff
@@ -8,6 +8,9 @@ class JustSelfAnnotation {
 @Self
 class ReturnType {
 
-    fun returnType(): Self = this as Self
+    fun returnType(): Self {
+        val res: Self  = this <!UNCHECKED_CAST!>as Self<!>
```

Comment:
Probably it should not be reported here

Reply:
[Fixed](https://github.com/JetBrains/kotlin/pull/4796/commits/e48bffce66d8bbb82e9e78eec70e5403edeabb50#diff-9a9c6248bd41f194f61edd9c48b946a7277e1d63a3dec533dc61e23184f32682R44)